lib/iris/tests/unit/lazy_data/test_get_fill_value.py

```diff
@@ -0,0 +1,75 @@
+# (C) British Crown Copyright 2017, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""Test the function :func:`iris._lazy data.get_fill_value`."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import dask.array as da
+import numpy as np
+import numpy.ma as ma
+
+from iris._lazy_data import as_lazy_data, get_fill_value
+
+
+class Test_get_fill_value(tests.IrisTest):
+    def setUp(self):
+        # Array shape and fill-value.
+        spec = [((2, 3, 4, 5), -1),  # 4d array
+                ((2, 3, 4), -2),     # 3d array
+                ((2, 3), -3),        # 2d array
+                ((2,), -4),          # 1d array
+                ((), -5)]            # 0d array
+        self.arrays = [np.empty(shape) for (shape, _) in spec]
+        self.masked = [ma.empty(shape, fill_value=fv) for (shape, fv) in spec]
+        self.lazy_arrays = [as_lazy_data(array) for array in self.arrays]
+        self.lazy_masked = [as_lazy_data(array) for array in self.masked]
+        # Add the masked constant case.
+        mc = ma.array([0], mask=True)[0]
```

Comment:
I think it would be better to specify a definite, non-default fill-value here, like all the other cases.
Just to make sure we're not getting a default instead.